### PR TITLE
Added ListObject and DeleteObject Config and Unit tests

### PR DESCRIPTION
**Pull request by athira-varghese**
https://github.ibm.com/Observability/dragonlog/issues/1360

### CODE REVIEW

**Review by adityacs — APPROVED** [review]
LGTM 

We may have to rebase based on which PR get's merged first. https://github.com/amankrsingh2000/loki/pull/5 also has overlapping changes

**Comment by adityacs** [comment]
Do not merge

File: pkg/storage/chunk/client/ibmcloud/cos_object_client_test.go [hunk 1]
```diff
@@ -81,3 +124,125 @@ func Test_COSConfig(t *testing.T) {
...
+		}
+		require.NoError(t, err)
+		sort.Slice(storage_obj, func(i, j int) bool {
+			return storage_obj[i].Key < storage_obj[j].Key
+		})
+		require.Equal(t, tt.storage_obj, storage_obj)
+	}
+}
+
+func (cosClient *mockCosClient) DeleteObjectWithContext(ctx context.Context, input *s3.DeleteObjectInput, opts ...request.Option) (*s3.DeleteObjectOutput, error) {
```
Comment (tareqmamari) [inline pkg/storage/chunk/client/ibmcloud/cos_object_client_test.go]: could you please put all the methods belong to mockCosClient together after the struct is defined?

File: pkg/storage/chunk/client/ibmcloud/cos_object_client_test.go [hunk 2]
```diff
@@ -81,3 +159,90 @@ func Test_COSConfig(t *testing.T) {
...
+		err = cosClient.DeleteObject(context.Background(), tt.key)
+		if tt.wantErr != nil {
+			require.Equal(t, tt.wantErr.Error(), err.Error())
+			continue
+		}
+		require.NoError(t, err)
+	}
+}
+
+func Test_List(t *testing.T) {
```
Comment (tareqmamari) [inline pkg/storage/chunk/client/ibmcloud/cos_object_client_test.go:367]: This test is failing. Could you please fix it?

File: pkg/storage/chunk/client/ibmcloud/cos_object_client_test.go [hunk 3]
```diff
@@ -81,3 +159,90 @@ func Test_COSConfig(t *testing.T) {
...
+		}
+		cosClient, err := NewCOSObjectClient(cosConfig, hedging.Config{})
+		require.NoError(t, err)
+		cosClient.cos = newMockCosClient()
+		err = cosClient.DeleteObject(context.Background(), tt.key)
+		if tt.wantErr != nil {
+			require.Equal(t, tt.wantErr.Error(), err.Error())
+			continue
+		}
+		require.NoError(t, err)
```
Comment (shahulsonhal) [inline pkg/storage/chunk/client/ibmcloud/cos_object_client_test.go]: Could you please add a get call and check the deleted object is no longer exist in bucket.

**Review by tareqmamari — APPROVED** [review]

**Pull request merged**